tropopause_find/tropopause_find.F90

```diff
@@ -0,0 +1,1464 @@
+
+    character(len=512), intent(out) :: errmsg
+    integer,            intent(out) :: errflg
+
+    errmsg = ' '
+    errflg = 0
+
+    ! Initialize the results to a missing value, so that the algorithms will
+    ! attempt to find the tropopause for all of them.
+    tropLev(:) = NOTFOUND
+    tropP(:) = fillvalue
+    tropT(:) = fillvalue
+    tropZ(:) = fillvalue
+
+    ! Try to find the tropopause using the primary algorithm.
+    if (primary /= TROP_ALG_NONE) then
+      call tropopause_findUsing(ncol, pver, lat, pint, pmid, t, zi, zm, phis, &
+                                calday, tropp_p_loc, tropp_days, &
+                                primary, tropLev, tropP=tropP, tropT=tropT, tropZ=tropZ, &
+                                hstobie_trop=hstobie_trop, hstobie_linoz=hstobie_linoz, hstobie_tropop=hstobie_tropop, & ! only for HYBSTOB
+                                errmsg=errmsg, errflg=errflg)
+    end if
+
+    if ((backup /= TROP_ALG_NONE) .and. any(tropLev(:) == NOTFOUND)) then
+      call tropopause_findUsing(ncol, pver, lat, pint, pmid, t, zi, zm, phis, &
+                                calday, tropp_p_loc, tropp_days, &
+                                backup, tropLev, tropP=tropP, tropT=tropT, tropZ=tropZ, &
+                                errmsg=errmsg, errflg=errflg)
+    end if
+
+  end subroutine tropopause_findWithBackup
+
+  ! Call the appropriate tropopause detection routine based upon the algorithm
+  ! specifed.
+  !
+  ! NOTE: It is assumed that the output fields have been initialized by the
+  ! caller, and only output values set to fillvalue will be detected.
+  subroutine tropopause_findUsing(ncol, pver, lat, pint, pmid, t, zi, zm, phis, &
+                                  calday, tropp_p_loc, tropp_days, &
+                                  algorithm, tropLev, tropP, tropT, tropZ, &
+                                  hstobie_trop, hstobie_linoz, hstobie_tropop, &
+                                  errmsg, errflg)
+
+    integer,         intent(in)         :: ncol          ! Number of atmospheric columns
+    integer,         intent(in)         :: pver          ! Number of vertical levels
+    real(kind_phys), intent(in)         :: lat(:)        ! Latitudes (radians)
+    real(kind_phys), intent(in)         :: pint(:,:)     ! Interface pressures (Pa)
+    real(kind_phys), intent(in)         :: pmid(:,:)     ! Midpoint pressures (Pa)
+    real(kind_phys), intent(in)         :: t(:,:)        ! Temperature (K)
+    real(kind_phys), intent(in)         :: zi(:,:)       ! Geopotential height above surface at interfaces (m)
+    real(kind_phys), intent(in)         :: zm(:,:)       ! Geopotential height above surface at midpoints (m)
+    real(kind_phys), intent(in)         :: phis(:)       ! Surface geopotential (m2 s-2)
+
+    real(kind_phys), intent(in)         :: calday        ! Day of year including fraction of day
+
+    ! Climatological tropopause pressures (Pa), (pcols,ntimes=12).
```

Comment:
As I was resolving comments, I saw this is comment which refers to pcols.  It (and any place else which refers to `pcols `in this code) should be replaced with `ncol`.

Reply:
Thank you, replaced with ncol!